packages/compass-aggregations/src/modules/saved-pipeline.ts

```diff
@@ -112,28 +112,33 @@ export const updatePipelineList =
 /**
  * Restore pipeline by an ID
  */
-export const openPipelineById = (
-  id: string
-): PipelineBuilderThunkAction<Promise<void>> => {
-  return async (dispatch, getState, { pipelineBuilder, pipelineStorage }) => {
+export const openStoredPipeline = (
+  pipelineData: StoredPipeline,
+  updatePreview = true
+): PipelineBuilderThunkAction<void> => {
+  return (dispatch, getState, { pipelineBuilder }) => {
     try {
-      const data = await pipelineStorage.load(id);
-      if (!data) {
-        throw new Error(`Pipeline with id ${id} not found`);
-      }
-      pipelineBuilder.reset(data.pipelineText);
+      pipelineBuilder.reset(pipelineData.pipelineText);
       dispatch({
         type: RESTORE_PIPELINE,
         stages: pipelineBuilder.stages,
         pipelineText: pipelineBuilder.source,
         pipeline: pipelineBuilder.pipeline,
         syntaxErrors: pipelineBuilder.syntaxError,
-        restoreState: data,
+        storedOptions: {
+          id: pipelineData.id,
+          name: pipelineData.name,
+          collationString: pipelineData.collationString,
+          comments: pipelineData.comments,
+          autoPreview: pipelineData.autoPreview,
+        },
```

Comment:
Instead of passing everything, I picked only what is being used by other reducers on restore